v3/src/components/graph/models/data-configuration-model.ts

```diff
@@ -144,6 +144,14 @@ export const DataConfigurationModel = types
     placeShouldShowClickHereCue(place: GraphPlace, tileHasFocus: boolean) {
       return this.placeAlwaysShowsClickHereCue(place) ||
         (this.placeCanShowClickHereCue(place) && tileHasFocus)
+    },
+    isCaseInSubPlot(subPlotKey: Record<string, string>, caseData: Record<string, any>) {
+      const numOfKeys = Object.keys(subPlotKey).length
+      let matchedValCount = 0
+      Object.keys(subPlotKey).forEach(key => {
+        if (!subPlotKey[key] || subPlotKey[key] === caseData[key]) matchedValCount++
```

Comment:
```suggestion
        if (subPlotKey[key] === caseData[key]) matchedValCount++
```
Do we still need the `!subPlotKey[key]` part of the test? I wouldn't think that it would occur at this point.

Reply:
@kswenson I think we do need this check, at least with the way things are right now. If I understand correctly, `!subPlotKey[key]` would be true when `!subPlotKey[key]` is equal to the empty string. In the latest commit I've changed it to `subPlotKey[key] === ""` to make it a bit more clear.

A key's value will be equal to the empty string when the associated attribute is numeric. In [adornments.tsx](https://github.com/concord-consortium/codap/blob/2e30125416e623b948c0645ccc64708c52d55ddb/v3/src/components/graph/adornments/adornments.tsx#L44-L54), when we set up the arrays used to iterate over when building the adornment grid, if an attribute is not categorical, we set the associated array to `[""]`. That way there's a single item to iterate over. I'm wondering now if we should revisit that idea since it's not self-evident a single empty string means the data isn't categorical. I think it also may be odd to do that mainly just to get the associated `for` loop to run once. Would it be worth trying to modify that approach so the arrays would be empty or undefined instead? Or is it reasonably straightforward as is?